python/cudf/cudf/core/series.py

```diff
@@ -2109,7 +2109,10 @@ def apply(self, func, convert_dtype=True, args=(), **kwargs):
         """
         if convert_dtype is not True:
             raise ValueError("Series.apply only supports convert_dtype=True")
-        return self._apply(func, _get_scalar_kernel, *args, **kwargs)
+
+        result = self._apply(func, _get_scalar_kernel, *args, **kwargs)
+        result.name = self.name
```

Comment:
Perhaps this can be done in `_apply`?

Reply:
I had considered this initially the problem being that `DataFrame` also inherits `_apply` from `IndexedFrame`, and `DataFrame` objects have no `name`.

Reply:
True, but you could add a `hasattr` check.

Reply:
I agree this is best handled in a post-processing step specific to `Series` rather than trying to check for attributes in a parent class that is not an effective model of its children. The current change in this PR (`result.name = self.name`) is fine with me.

Reply:
We don't use `_apply()` internally, but if we did, I can see the benefit of correct name handling. I don't feel strongly here. Defer to Mike on the final word

Reply:
:100: agree with using inheritance here. Thanks @shwina!